core/src/main/java/com/zextras/carbonio/user_management/services/UserService.java

```diff
@@ -45,6 +49,34 @@ private UserInfo createUserInfo(GetAccountInfoResponse accountInfo) {
     return userInfo;
   }
 
+  public Response getUsers(List<UUID> userIds, String token) {
+    return (!userIds.isEmpty()) ? Response.status(Status.BAD_REQUEST).build()
+      : Response.ok().entity(
+          userIds.stream().map(userUuid -> {
+            System.out.println("Requested: " + userUuid);
+            UserInfo userInfo = cacheManager.getUserByIdCache().getIfPresent(userUuid);
+
+            if (userInfo == null) {
+              try {
+                GetAccountInfoResponse accountInfo = SoapClient
```

Comment:
I suppose that the existing bulk soap operation does not work as expected so you are fall back to an iterative approach. Can you confirm?

Reply:
Confirmed, the bulk requests a contactId instead of a userId and the returned parameters are not manageable